src/util_gen.cpp

```diff
@@ -470,6 +477,14 @@ bool Util::isValidFd(int fd)
   return _real_fcntl(fd, F_GETFL, 0) != -1;
 }
 
+// On failure, returns 0, with errno == EFAULT
+bool Util::isReadableMemoryAddress(const char *addr) {
+  // FIXME: We wanted to call libc:open(), but _real_open goes to second wrapper
+  int fd = _real_syscall(SYS_open, "/dev/null", O_WRONLY);
+  int rc = write(fd, addr, 1);
```

Comment:
Doing a write of "0" bytes is not dangerous. It's a well known trick and is used quite generously.  The POSIX standard documents this behavior too. I would strongly suggest to use this instead of doing an open and close.